### PR TITLE
PowerShell mode for CodeMirror

### PR DESCRIPTION
**Pull request by ashmind**
This is a PowerShell mode for CodeMirror.
I've done it a while ago, so I don't remember all details, but the following should work well:

1. Various literals
2. Variables, including `${}`
3. Various operators
4. String interpolation, including nested string interpolation
5. Builtins (known variables and functions)

And I've just added some tests (around 94% code coverage).

Let me know if there are any issues.

### CODE REVIEW

**Comment by Dominator008** [comment]
@ashmind Can you squash the commits? Thanks.

File: mode/powershell/test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  [
+    '(', ')', '[[', ']]', '{', '}', ',', '`', ';', '.'
+  ].forEach(function(punctuation) {
+    MT("punctuation_" + punctuation.replace(/^[\[\]]/,''), "[punctuation " + punctuation + "]");
+  });
+
+  MT('keyword', "[keyword if]");
+
+  MT('call_builtin', "[builtin Get-ChildItem]");
+})();
```
Comment (Dominator008) [inline mode/powershell/test.js]: This needs an additional newline afterwards.

**Comment by ashmind** [comment]
@Dominator008 I thought about squashing them, but wouldn't that eliminate @benpmiller and @VapidWorx from the history?

**Comment by Dominator008** [comment]
@ashmind Maybe squash into three / four commits?

**Comment by ashmind** [comment]
@Dominator008 OK I'll try that.

**Comment by ashmind** [comment]
@Dominator008 See #3911.

**Pull request closed without merging**